stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt

```diff
@@ -197,43 +234,57 @@ internal fun BoxScope.DefaultMessagesHelperContent(
 
     val offset = messagesState.focusedMessageOffset.collectAsState()
 
-    LaunchedEffect(
-        newMessageState,
-        firstVisibleItemIndex,
-        focusedItemIndex,
-        offset.value
-    ) {
-        if (focusedItemIndex != -1 && !lazyListState.isScrollInProgress) {
+    LaunchedEffect(newMessageState, focusedItemIndex, offset.value, scrollToStartState) {
+
+        if (focusedItemIndex != -1 &&
+            !lazyListState.isScrollInProgress &&
+            scrollToStartState is ScrollToFocusedMessage
+        ) {
             coroutineScope.launch {
                 lazyListState.scrollToItem(focusedItemIndex, offset.value ?: 0)
             }
         }
 
+        if (scrollToStartState == ScrollToNewestMessages) {
+            lazyListState.scrollToItem(0)
+        }
+
         when {
-            !lazyListState.isScrollInProgress && newMessageState == Other &&
-                firstVisibleItemIndex < 3 -> coroutineScope.launch {
-                lazyListState.animateScrollToItem(0)
+            focusedItemIndex == -1 &&
+                !lazyListState.isScrollInProgress &&
```

Comment:
This is extremely complex - can we pull this out in a function and then call this value something meaningful?